avocado/core/test.py

```diff
@@ -133,6 +133,98 @@ def str_filesystem(self):
                                  % (self.str_uid, str(self)))
 
 
+def get_file_data_dir(test):
+    """
+    Returns the path to the file-level data directory
+
+    For test a test file hosted at /usr/share/avocado/tests/sleeptest.py
+    the file-level data dir is /usr/share/avocado/tests/sleeptest.py.data.
+
+    Note that this directory has no specific relation to the test
+    name, only to the file that contains the test.  It can be used to
+    host data files that are generic enough to be used for all tests
+    contained in a given test file.
+
+    This function is currently a thin wrapper (almost an alias) for
+    :attr:`avocado.core.test.Test.datadir`.  The idea is just to give
+    consistent naming across the various data dir, and prepare for the
+    removal of :attr:`avocado.core.test.Test.datadir`.
+
+    :param test: the test to locate its file-level data directory
+    """
+    # TODO: on removal of Test.datadir, move its code over here
+    return test.datadir
```

Comment:
Well this one is currently tied to `self.filename`, but do you think people might need to supply different base-dir instead? For example the `robot` plugin extracts the `filename` from the test name, therefor it'd work well for that one, but what if there is a test type which does not use filename, or wants to specify different base-path for datadir than `filename + "data"`?

Perhaps this should always stay as a symlink to `test.datadir` which in the future might become `test.base_datadir`. On the other hand we can just say the `filename + ".data"` is mandatory and use it as defined in your version...

Reply:
The idea here is to start with something users can relate to.  Tests already have a `datadir`, and this concept is still valid.

A test type, say, `VT`, could in theory define their `datadir` implementation, and it'd be respected here.

If we want to make a test's datadir even more flexible, we could, but I think this is enough at this point.